File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/InsurancePlan.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class InsurancePlan : DomainResource, IResource
{
    public Base.ResourceReference[]? CoverageArea { get; set; }
    public string? Name { get; set; }
    public InsurancePlan_Coverage[]? Coverage { get; set; }
    public Base.CodeableConcept[]? Type { get; set; }
    public string[]? Alias { get; set; }
    public string? Status { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.ResourceReference? AdministeredBy { get; set; }
    public Base.ResourceReference? OwnedBy { get; set; }
    public Base.ResourceReference[]? Network { get; set; }
    public Base.Period? Period { get; set; }
    public InsurancePlan_Plan[]? Plan { get; set; }
    public Base.ResourceReference[]? Endpoint { get; set; }
    public InsurancePlan_Contact[]? Contact { get; set; }

    public class InsurancePlan_Coverage_Benefit_Limit : BackboneElement
    {
        public Base.CodeableConcept? Code { get; set; }
        public Base.Quantity? Value { get; set; }
    }

    public class InsurancePlan_Coverage_Benefit : BackboneElement
    {
        public required Base.CodeableConcept Type { get; set; }
        public InsurancePlan_Coverage_Benefit_Limit[]? Limit { get; set; }
        public string? Requirement { get; set; }
    }

    public class InsurancePlan_Coverage : BackboneElement
    {
        public required Base.CodeableConcept Type { get; set; }
        public required InsurancePlan_Coverage_Benefit[] Benefit { get; set; }
        public Base.ResourceReference[]? Network { get; set; }
    }

    public class InsurancePlan_Plan_GeneralCost : BackboneElement
    {
        public Base.Money? Cost { get; set; }
        public Base.CodeableConcept? Type { get; set; }
        public string? Comment { get; set; }
        public string? GroupSize { get; set; }
    }

    public class InsurancePlan_Plan_SpecificCost_Benefit_Cost : BackboneElement
    {
        public required Base.CodeableConcept Type { get; set; }
        public Base.Quantity? Value { get; set; }
        public Base.CodeableConcept[]? Qualifiers { get; set; }
        public Base.CodeableConcept? Applicability { get; set; }
    }

    public class InsurancePlan_Plan_SpecificCost_Benefit : BackboneElement
    {
        public InsurancePlan_Plan_SpecificCost_Benefit_Cost[]? Cost { get; set; }
        public required Base.CodeableConcept Type { get; set; }
    }

    public class InsurancePlan_Plan_SpecificCost : BackboneElement
    {
        public InsurancePlan_Plan_SpecificCost_Benefit[]? Benefit { get; set; }
        public required Base.CodeableConcept Category { get; set; }
    }

    public class InsurancePlan_Plan : BackboneElement
    {
        public Base.CodeableConcept? Type { get; set; }
        public Base.ResourceReference[]? Network { get; set; }
        public Base.Identifier[]? Identifier { get; set; }
        public InsurancePlan_Plan_GeneralCost[]? GeneralCost { get; set; }
        public Base.ResourceReference[]? CoverageArea { get; set; }
        public InsurancePlan_Plan_SpecificCost[]? SpecificCost { get; set; }
    }

    public class InsurancePlan_Contact : BackboneElement
    {
        public Base.HumanName? Name { get; set; }
        public Base.Address? Address { get; set; }
        public Base.CodeableConcept? Purpose { get; set; }
        public Base.ContactPoint[]? Telecom { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/InsurancePlan.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Address
from base import Period
from base import CodeableConcept
from base import ContactPoint
from base import Quantity
from base import HumanName
from base import DomainResource
from base import Money
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class InsurancePlanCoverageBenefitLimit(BackboneElement):
    code: Optional[CodeableConcept] = None
    value: Optional[Quantity] = None

@dataclass(kw_only=True)
class InsurancePlanCoverageBenefit(BackboneElement):
    type: CodeableConcept
    limit: Optional[List[InsurancePlanCoverageBenefitLimit]] = field(default_factory=list)
    requirement: Optional[str] = None

@dataclass(kw_only=True)
class InsurancePlanCoverage(BackboneElement):
    type: CodeableConcept
    benefit: list[InsurancePlanCoverageBenefit] = field(default_factory=list)
    network: Optional[List[Reference]] = field(default_factory=list)

@dataclass(kw_only=True)
class InsurancePlanPlanGeneralCost(BackboneElement):
    cost: Optional[Money] = None
    type: Optional[CodeableConcept] = None
    comment: Optional[str] = None
    group_size: Optional[int] = None

@dataclass(kw_only=True)
class InsurancePlanPlanSpecificCostBenefitCost(BackboneElement):
    type: CodeableConcept
    value: Optional[Quantity] = None
    qualifiers: Optional[List[CodeableConcept]] = field(default_factory=list)
    applicability: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class InsurancePlanPlanSpecificCostBenefit(BackboneElement):
    cost: Optional[List[InsurancePlanPlanSpecificCostBenefitCost]] = field(default_factory=list)
    type: CodeableConcept

@dataclass(kw_only=True)
class InsurancePlanPlanSpecificCost(BackboneElement):
    benefit: Optional[List[InsurancePlanPlanSpecificCostBenefit]] = field(default_factory=list)
    category: CodeableConcept

@dataclass(kw_only=True)
class InsurancePlanPlan(BackboneElement):
    type: Optional[CodeableConcept] = None
    network: Optional[List[Reference]] = field(default_factory=list)
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    general_cost: Optional[List[InsurancePlanPlanGeneralCost]] = field(default_factory=list)
    coverage_area: Optional[List[Reference]] = field(default_factory=list)
    specific_cost: Optional[List[InsurancePlanPlanSpecificCost]] = field(default_factory=list)

@dataclass(kw_only=True)
class InsurancePlanContact(BackboneElement):
    name: Optional[HumanName] = None
    address: Optional[Address] = None
    purpose: Optional[CodeableConcept] = None
    telecom: Optional[List[ContactPoint]] = field(default_factory=list)

@dataclass(kw_only=True)
class InsurancePlan(DomainResource):
    coverage_area: Optional[List[Reference]] = field(default_factory=list)
    name: Optional[str] = None
    coverage: Optional[List[InsurancePlanCoverage]] = field(default_factory=list)
    type: Optional[List[CodeableConcept]] = field(default_factory=list)
    alias: Optional[List[str]] = field(default_factory=list)
    status: Optional[str] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    administered_by: Optional[Reference] = None
    owned_by: Optional[Reference] = None
    network: Optional[List[Reference]] = field(default_factory=list)
    period: Optional[Period] = None
    plan: Optional[List[InsurancePlanPlan]] = field(default_factory=list)
    endpoint: Optional[List[Reference]] = field(default_factory=list)
    contact: Optional[List[InsurancePlanContact]] = field(default_factory=list)